src/wallet/wallet.cpp

```diff
@@ -3618,14 +3714,41 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
         walletInstance->SetMaxVersion(nMaxVersion);
     }
 
+
+    std::string hdWatchOnly = GetArg("-hdwatchonly", "");
+    CExtPubKey extPubKey;
+    if (!hdWatchOnly.empty()) {
+        CBitcoinExtPubKey bitcoinExtPubKey;
+        auto extpubKeyPrefix = chainparams.Base58Prefix(CChainParams::Base58Type::EXT_PUBLIC_KEY);
```

Comment:
Interesting thing: https://github.com/bitcoin/bitcoin/blob/master/src/base58.h#L160 it does not check the actual prefix.
I think we should dump this ctor, it is using `Params()` and does not check the prefix.